crates/parser2/src/parser/expr.rs

```diff
@@ -11,6 +11,10 @@ use super::{
 pub fn parse_expr<S: TokenStream>(parser: &mut Parser<S>) -> bool {
     parse_expr_with_min_bp(parser, 0, true)
 }
+pub fn parse_lhs_aug<S: TokenStream>(parser: &mut Parser<S>) -> bool {
+    //include Lp and Lb ?
+    parse_expr_with_min_bp(parser, 146, true)
```

Comment:
I did some work on `augassign`; Is this `bp` acceptable ? (currently it takes `dot`, `Lparen`, `Lbrack`, and 'Lt')

Reply:
`bp` seems fine. 

However, I've started considering changing the specifications of the assignment. Clearly, it'd be better to define `assign` and `augassign` as expressions rather than statements. 
Because:
1. We can implement a trait `AddAssign` or something like that.
2. Emitting a good error message in the parsing phase is difficult when they are defined as statements. e.g., please consider `x + 1 += 1`.

This is my fault. I didn't fully consider it, even though I felt I should've when I implemented parser2. Sorry.

Reply:
So, I think we could take two options.
1. Merge this PR as is, then someone could change the specification, or
2. You could work on the change in this PR if you are willing to help us

Anyways, I'm sorry that I didn't consider it fully and gave insufficient suggestions.

Reply:
@sbillig Any thoughts?

Reply:
No worries, I can start working on it.

Reply:
> No worries, I can start working on it.

Ok, thanks!
 
the rough sketch for implementing this would be
1. Modify the parser so that it can parse them as expressions.
2. Change the HIR definition for `Assign` and `AugAssign`
3. Rewrite the HIR lowering and remove the unnecessary desugaring related types(e.g., `hir::span::DesugaredOrigin::AugAssign`)

Reply:
Please feel free to let me know if you find any troubles.

Reply:
hey Yoshi, 2 questions: 
1) the `push_expr` receives an `origin` so maybe we shouldn't delete the `hir::span::DesugaredOrigin::AugAssign` ??
![image](https://github.com/ethereum/fe/assets/73394405/a235b57c-9e6c-4487-ae30-675af740b236)

2) the `aug_assign` and `assign` are going to be parsed in the `parse_atom_expr` correct ?? if so, is it ok to pass the `checkpoint` as `null` ??  sth along the lines of:
![image](https://github.com/ethereum/fe/assets/73394405/d81c98c1-8c6b-41c2-902e-3d7ca180926e)

Reply:
> the push_expr receives an origin so maybe we shouldn't delete the hir::span::DesugaredOrigin::AugAssign ??

We don't need `desugar_aug_assign` and `DesugaredOrigin::AugAssign` because we no longer need to desugar them. Instead, you need to modify `hir::expr::Expr` like the one below.
```rust
pub enum Expr {
    Assign(ExprId, ExprId),
    AugAssign(ExprId, ExprId, BinOp),
    ...
}
```

Then you could lower `ast::Assign` and `ast::AugAssign` in `lower_expr`.

Reply:
> the aug_assign and assign are going to be parsed in the parse_atom_expr correct ??

I don't think you need to modify `parse_expr_atom`.  Conceptually, what you need to do is the same that `BinExprScope` does. The one difficult thing is we don't have tokens to represent AugAssign operators(e.g., `+=`). 
This would be solved in two different ways. 
1. Just modify `SyntaxKind` and accept them as an atom kind (Please refer to `SyntaxKind`).
2. Modify `infix_binding_power` and `bump_bin_op` to handle them, then if `bump_bin_op` indicates that the binary operator is assign or aug assign, call `set_kind` in `BinExprScope::parse` method. Please refer to `ParenScope::parse` for `set_kind` method.

I think option `1.` is easier to implement, but I prefer option `2.` because it allows us to extract the span of the binary operation part in augassign op(e.g., span for `+` in `+=`)